invenio_communities/communities/records/systemfields/children.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2023 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Children system field."""
+
+from invenio_records.systemfields import SystemField
+
+
+class Children:
+    """Children object."""
+
+    def __init__(self, allow=None):
+        self._allow = allow or False
+
+    @property
+    def allow(self):
+        """Get the allow."""
+        return self._allow
+
+    @allow.setter
+    def allow(self, value):
+        """Set the allow."""
+        if value is not None:
+            assert isinstance(value, bool)
+        self._allow = value
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a Children object from a dictionary."""
+        return cls(allow=data.get("allow"))
+
+    def dump(self):
+        """Dump the Children object to a dictionary."""
+        return {"allow": self.allow}
+
+
+class ChildrenField(SystemField):
+    """System field for children of a community."""
+
+    children_obj_class = Children
+
+    def __init__(self, key="children", children_obj_class=None):
+        """Create a new ChildrenField instance."""
+        self.children_obj_class = children_obj_class or self.children_obj_class
+        super().__init__(key=key)
+
+    def obj(self, instance):
+        """Get the Children object."""
+        obj = self._get_cache(instance)
+        if obj is not None:
+            return obj
+
+        data = self.get_dictkey(instance)
+        if data:
+            obj = self.children_obj_class.from_dict(data)
+        else:
+            obj = self.children_obj_class()
+
+        self._set_cache(instance, obj)
+        return obj
+
+    def set_obj(self, record, obj):
+        """Set the Children object."""
+        # We accept both dicts and Children class objects.
+        if isinstance(obj, dict):
+            obj = self.children_obj_class.from_dict(obj)
+
+        assert isinstance(obj, self.children_obj_class)
+
+        # We do not dump the object until the pre_commit hook
+        self._set_cache(record, obj)
+
+    def __get__(self, record, owner=None):
+        """Get the record's Children object."""
+        if record is None:
+            # access by class
+            return self
+
+        # access by object
+        return self.obj(record)
+
+    def __set__(self, record, obj):
+        """Set the records Children object."""
+        self.set_obj(record, obj)
+
+    def pre_commit(self, record):
```

Comment:
comment/shelve: we should have probably done the same with the parent field, i.e. have a `pre_commit` to manage how/when the field is persisted in the record